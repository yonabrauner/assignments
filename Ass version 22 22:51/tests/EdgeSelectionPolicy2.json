{
  "parties": [
    {
      "name": "a",
      "mandates": 10,
      "join_policy": "L"
    },
    {
      "name": "b",
      "mandates": 10,
      "join_policy": "L"
    },
    {
      "name": "c",
      "mandates": 10,
      "join_policy": "L"
    },
    {
      "name": "d",
      "mandates": 10,
      "join_policy": "L"
    },
    {
      "name": "e",
      "mandates": 80,
      "join_policy": "L"
    }
  ],
  "graph": [
    [0, 1, 1, 1, 1],
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0],
    [1, 0, 0, 0, 0]
  ],
  "agents": [
    {
      "party_id": 0,
      "selection_policy": "E"
    }
  ]
}
