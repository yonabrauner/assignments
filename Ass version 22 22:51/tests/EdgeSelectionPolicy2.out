[
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 10,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 10,
                "name": "b",
                "state": "Waiting"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "c",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "d",
                "state": "Waiting"
            },
            {
                "id": 4,
                "mandates": 80,
                "name": "e",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 10,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 10,
                "name": "b",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "c",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "d",
                "state": "Waiting"
            },
            {
                "id": 4,
                "mandates": 80,
                "name": "e",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 10,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 10,
                "name": "b",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "c",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "d",
                "state": "Waiting"
            },
            {
                "id": 4,
                "mandates": 80,
                "name": "e",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            }
        ],
        "coalitions": [
            {
                "mandates": 10,
                "parties": [
                    0
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 10,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 10,
                "name": "b",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "c",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "d",
                "state": "CollectingOffers"
            },
            {
                "id": 4,
                "mandates": 80,
                "name": "e",
                "state": "Waiting"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 20,
                "parties": [
                    0,
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 10,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 10,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "c",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "d",
                "state": "CollectingOffers"
            },
            {
                "id": 4,
                "mandates": 80,
                "name": "e",
                "state": "CollectingOffers"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            },
            {
                "id": 2,
                "partyId": 2
            }
        ],
        "coalitions": [
            {
                "mandates": 30,
                "parties": [
                    0,
                    1,
                    2
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 10,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 10,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "c",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "d",
                "state": "CollectingOffers"
            },
            {
                "id": 4,
                "mandates": 80,
                "name": "e",
                "state": "CollectingOffers"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            },
            {
                "id": 2,
                "partyId": 2
            },
            {
                "id": 3,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 40,
                "parties": [
                    0,
                    1,
                    2,
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 10,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 10,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "c",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "d",
                "state": "Joined"
            },
            {
                "id": 4,
                "mandates": 80,
                "name": "e",
                "state": "CollectingOffers"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 1
            },
            {
                "id": 2,
                "partyId": 2
            },
            {
                "id": 3,
                "partyId": 3
            },
            {
                "id": 4,
                "partyId": 4
            }
        ],
        "coalitions": [
            {
                "mandates": 120,
                "parties": [
                    0,
                    1,
                    2,
                    3,
                    4
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 1,
                "weight": 1
            },
            {
                "from": 0,
                "to": 2,
                "weight": 1
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 0,
                "to": 4,
                "weight": 1
            },
            {
                "from": 1,
                "to": 0,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 4,
                "to": 0,
                "weight": 1
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 10,
                "name": "a",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 10,
                "name": "b",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 10,
                "name": "c",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 10,
                "name": "d",
                "state": "Joined"
            },
            {
                "id": 4,
                "mandates": 80,
                "name": "e",
                "state": "Joined"
            }
        ]
    }
]
